skiko/src/jvmMain/kotlin/org/jetbrains/skiko/Library.kt

```diff
@@ -128,3 +145,31 @@ internal class LibraryTestImpl() {
         return bitmap._ptr
     }
 }
+
+/**
+ * Modifications to the [SkikoProperties.dataPath] shall be locked using this file, to avoid
+ * multiple processes acting on the files (e.g. racing to unpack the native library)
+ */
+private val SkikoProperties.dataPathLockfile get() = File(this.dataPath).resolve(".lock")
+
+private val dataPathMutex = ReentrantLock()
+
+/**
+ * Ensures that only single process & single thread can modify the [SkikoProperties.dataPath].
+ * Uses the [dataPathLockfile] as filelock to synchronize with other processes
+ */
+internal inline fun lockSkikoDataDirectory(action: () -> Unit) {
+    if (dataPathMutex.isHeldByCurrentThread) action()
```

Comment:
It is better to remove this check for simplicity - entering this function should be protected by the lock in the `load` function.

Reply:
I looked at it as a private function of this file. The current code with a doc looks fine to me, the previous one also looked fine but had more complexity.